### PR TITLE
chore: 🤖 add role principal read ability

### PR DESCRIPTION
**Pull request by ghost**
Adds an ability to check read permission on a role principal.  This is a convenience since principals are polymorphic and would otherwise require a type lookup before the read ability check.

Usage examples:
- `can 'readPrincipal role' user`
- `can 'readPrincipal role' group`
- `can 'readPrincipal role' managedGroup`

### CODE REVIEW

File: addons/api/addon/abilities/role.js [hunk 1]
```diff
@@ -26,4 +35,22 @@ export default class RoleAbility extends ModelAbility {
...
+   * Returns the ability check for a principal model instance.
+   * Valid role principal models are: User, Group, and ManagedGroup.
+   * Throws an error if the principal type is invalid.
+   * @type {boolean}
+   * @throws {InvalidRolePrincipalTypeError}
+   */
+  get canReadPrincipal() {
+    const type = this.model.constructor.modelName;
+    const typeIsValid =
+      type === 'user' || type === 'group' || type === 'managed-group';
```
Comment (DhariniJeeva) [inline addons/api/addon/abilities/role.js:49]:
```suggestion
      type === 'user' || type === 'group' || type === 'managed group';
```

File: addons/api/addon/abilities/role.js [hunk 2]
```diff
@@ -26,4 +35,22 @@ export default class RoleAbility extends ModelAbility {
...
+    if (!typeIsValid)
+      throw new InvalidRolePrincipalTypeError(
+        `Expected a role principal of type 'user', 'group', or 'managed-group'.  Got '${type}'.`
```
Comment (DhariniJeeva) [inline addons/api/addon/abilities/role.js:52]:
```suggestion
        `Expected a role principal of type 'user', 'group', or 'managed group'.  Got '${type}'.`
```

File: addons/api/addon/abilities/role.js (hunk 1 above)
Comment (randallmorey) [reply addons/api/addon/abilities/role.js:49]: The input for this is an actual Ember Data model instance, in which the model name is `managed-group`.

File: addons/api/addon/abilities/role.js (hunk 2 above)
Comment (randallmorey) [reply addons/api/addon/abilities/role.js:52]: The input for this is an actual Ember Data model instance, in which the model name is `managed-group`.

File: .github/workflows/oss-ent-merge-trigger.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Trigger OSS to Enterprise Merge
```
Comment (calcaide) [inline .github/workflows/oss-ent-merge-trigger.yml:1]: Is this entire file part of this PR? To me seems out of context. Could we have a separate PR just for the GH workflow?

File: addons/api/addon/models/role.js [hunk 4]
```diff
@@ -75,7 +85,7 @@ export default class RoleModel extends GeneratedRoleModel {
         .queryBy('user', { id: ids }, { scope_id: 'global', recursive: true })
         .then((models) => models.map((model) => model));
     }
-
+    //this.store.query('user', { scope_id: 'global', recursive: true });
```
Comment (calcaide) [inline addons/api/addon/models/role.js:88]: Is this left in purpose?

File: addons/api/addon/models/role.js [hunk 5]
```diff
@@ -103,6 +113,43 @@ export default class RoleModel extends GeneratedRoleModel {
     return resolve([]);
   }
 
+  /**
+   * A promise that resolves to an array of Group instances.
+   * When calling this getter, be sure to await resolution
+   * before interacting with the results.
+   * @type {Promise[GroupModel]}
+   */
+  get managedGroups() {
```
Comment (calcaide) [inline addons/api/addon/models/role.js:122]: If we are aiming for an async function (based on the coment above) shouldn't we define this function as `async`?

**Pull request closed without merging**